agent/worker.go

```diff
@@ -42,17 +50,19 @@ type worker struct {
 	db        *bolt.DB
 	executor  exec.Executor
 	listeners map[*statusReporterKey]struct{}
+	secrets   *Secrets
```

Comment:
Oh cool - I thought it had to be a pointer in order to call functions that had pointer receivers.  Thanks